src/backend/distributed/planner/multi_master_planner.c

```diff
@@ -303,7 +304,7 @@ BuildAggregatePlan(Query *masterQuery, Plan *subPlan)
 		 * see nodeAgg.c:build_pertrans_for_aggref(). In that case we use
 		 * sorted agg strategy, otherwise we use hash strategy.
 		 */
-		if (!groupingIsHashable || hasDistinctAggregate)
+		if (!enable_hashagg || !groupingIsHashable || hasDistinctAggregate)
 		{
 			if (!groupingIsSortable)
 			{
```

Comment:
following error message is now incorrect.  
```
if (!enable_hashagg && groupingIsHashable && !hasDistinctAggregate)
``
setting ```enable_hasagg``` may get it to work. 
Perhaps we could provide this as hint.

Reply:
But the default for `enable_hasagg` is already on.

Reply:
I meant hint could be "setting enable_hashagg to true may help ...."